### DISCUSSION TITLE
twitter/x posts not captured - screenshots taken before the post is fully loaded,  need help

### DISCUSSION THREAD

**Discussion by gapetto** [ArchiveBox/ArchiveBox#1490] [Q&A]
So it looks like tweet posts they are captured as a single file, but pdf or screenshot don't work. The resulting image has just spinner icon, which indicates that post was not yet loaded before the screenshot was taken. I looked in configuration options but no option seems to help with this issue. Any ideas? I guess we would need something like delay before taking screenshot option. 

![image](https://github.com/user-attachments/assets/726a8cc6-b5a9-4a4b-8a0c-4115cffb49b6)

**Comment by pirate** [comment]
That's normal for big tech social media sites, that's why we use a ton of different methods to capture redundantly, usually a few will fail but others will work.

There is a CHROME_ARGS config option coming out in v0.8 that will allow setting a loading delay before screenshots. 

I do bot-blocking evasion / more advanced JS archiving stuff for paying clients, but it takes a *ton* of work to maintain so I haven't open sourced all of it yet:
https://docs.monadical.com/s/archivebox-consulting-services
